### PR TITLE
[buck-yeh-bux] add port

### PR DESCRIPTION
**Pull request by buck-yeh**
`bux` is a supplemental static library gathering whatever are seen required in sense of general purpose but not directly supported from Modern C++.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Successfully install the following and it is ok to support only them, **Yes**
  ~~~powershell
  vcpkg install bux:x64-windows
  vcpkg install bux:x64-windows-static
  vcpkg install bux:x64-windows-static-md
  vcpkg install bux:x86-windows
  vcpkg install bux:x86-windows-static
  ~~~

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Comment by autoantwort** [comment]
It is only supported on windows and not on uwp, linux and mac?

**Comment by buck-yeh** [comment]
> It is only supported on windows and not on uwp, linux and mac?

Yes, right now `bux` only manages to install on Windows. On ArchLinux, `bux` can be installed from the [AUR](https://aur.archlinux.org/packages/bux/). Packages for Ubuntu, Fedora are planned.

**Comment by buck-yeh** [comment]
FWIW, I run `./vcpkg install bux:x64-linux` in the fork directory on ArchLinux and it passes.

**Comment by dg0yt** [comment]
Does project 'bux' have any external users or contributors, or is it a personal project?

**Comment by buck-yeh** [comment]
> Does project 'bux' have any external users or contributors, or is it a personal project?

No, there is just  me.

**Comment by buck-yeh** [comment]
[CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) updated

File: ports/bux/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "bux",
+  "version": "1.4.1",
+  "port-version": 1,
```
Comment (mathisloge) [inline ports/bux/vcpkg.json:4]: You are adding a new port. there is no need to add a port-version:1. 
Each PR should only add one new version per port.

Comment (buck-yeh) [reply ports/bux/vcpkg.json]: Thank you! Fixed & queued for approval.

Comment (mathisloge) [reply ports/bux/vcpkg.json]: make sure that the referenced version git-tree is correct.

you can run `vcpkg x-add-version bux --overwrite-version` to make this automatically.

Comment (buck-yeh) [reply ports/bux/vcpkg.json]: It works!

File: ports/bux/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO buck-yeh/bux
+    REF 4636183159f74f350238e6ee59583b34f7e7aca9 # v1.4.1
+    SHA512 319cd15c73ca469849cbc364635d7147cfa3fed57d6043b0cb2b52d504a253645518ec6254ddba51fc8cc32cc89e345458530628b331629cabce5b64d8723a6a
+    HEAD_REF main
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/bux/portfile.cmake:11]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/bux/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+)
+
+vcpkg_install_cmake()
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/bux RENAME copyright)
```
Comment (JonLiu1993) [inline ports/bux/portfile.cmake:16]:
```suggestion
vcpkg_cmake_install()
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

File: scripts/ci.baseline.txt [hunk 4]
```diff
@@ -71,6 +71,10 @@ breakpad:arm64-windows=fail
 bullet3:arm64-windows=fail
 bullet3:arm-uwp=fail
 bullet3:x64-uwp=fail
+bux:x64-uwp=fail
+bux:arm-uwp=fail
+bux:x64-linux=fail
+bux:x64-osx=fail
```
Comment (JonLiu1993) [inline scripts/ci.baseline.txt:77]: If the upstream port itself does not support it, please indicate in vcpkg.json and portfile.cmake and delete these lines

File: ports/bux/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,16 @@
+vcpkg_from_github(
```
Comment (JonLiu1993) [inline ports/bux/portfile.cmake:1]:
```suggestion
vcpkg_fail_port_install(ON_ARCH "arm" "arm64")
vcpkg_fail_port_install(ON_TARGET "linux" "uwp" "osx")
vcpkg_from_github(
```

File: ports/bux/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "bux",
+  "version": "1.4.1",
+  "description": "A supplemental C++ library with functionalities not directly supported from Modern C++ standard.",
+  "homepage": "https://github.com/buck-yeh/bux",
```
Comment (JonLiu1993) [inline ports/bux/vcpkg.json:5]:
```suggestion
  "homepage": "https://github.com/buck-yeh/bux",
  "supports": "windows",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
make sure that the referenced version git-tree is correct.

you can run vcpkg x-add-version bux --overwrite-version to make this automatically.

File: ports/bux/vcpkg.json [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+  "supports": "windows",
+  "dependencies": [
+    "fmt"
+  ]
```
Comment (JonLiu1993) [inline ports/bux/vcpkg.json:9]:
```suggestion
  "dependencies": [
    "fmt",
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
  ]
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@buck-yeh ,Please take a look:
````
CMake Error at scripts/cmake/vcpkg_fail_port_install.cmake:96 (message):
  Architecture 'arm64' not supported by bux!
````

File: scripts/ci.baseline.txt (hunk 4 above)
Comment (JonLiu1993) [inline scripts/ci.baseline.txt:77]:
```suggestion
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: scripts/ci.baseline.txt (hunk 4 above)
Comment (autoantwort) [reply scripts/ci.baseline.txt:77]: Afaik upstream support this, but it fails in the pipeline because the compilers are not c++20 capable

**Comment by buck-yeh** [comment]
What should I do now?

**Review by JonLiu1993 — APPROVED** [review]

File: ports/bux/vcpkg.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "bux",
```
Comment (ras0219-msft) [inline ports/bux/vcpkg.json:2]:
```suggestion
  "name": "buck-yeh-bux",
```
`bux` is a very short name -- could you please rename the port to make it clear what library is referred to?

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
LGTM after renaming. Thanks for the PR and neat new library!

File: ports/bux/vcpkg.json (hunk 8 above)
Comment (buck-yeh) [reply ports/bux/vcpkg.json]: Changes pushed

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**